amaranth/sim/core.py

```diff
@@ -137,11 +134,10 @@ def wrapper():
                     try:
                         result = yield command
                         exception = None
-                        yield object.__new__(Settle)
                     except Exception as e:
                         result = None
                         exception = e
-        self._engine.add_coroutine_process(wrapper, default_cmd=None)
+        self._engine.add_testbench_process(wrapper)
```

Comment:
I believe this wrapper is now entirely redundant; all it does is filtering `Settle`, which can be done in the sim loop instead (which is now separate for testbenches)

Reply:
Huh? The main sim loop doesn't process commands (or is even aware of their presence).

Reply:
er, I meant the loop in `PyCoroProcess.run`

Reply:
Oh yeah let me fix that